app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapter.java

```
@@ -174,6 +184,21 @@ public void updateContact(final Contact oldContact, final Contact newContact) th
      * {@inheritDoc}
      */
     @Override
+    public Contact getContactWithName(final String contactName) throws IOException {
+        final List<Contact> contacts = getAllContacts();
+        for (final Contact contact: contacts) {
+            if(contact.getName().equals(contactName)) {
+                return contact;
+            }
+        }
+
+        return null;
```

Maybe it's better to throw an exception instead of returning null